pkg/types/openstack/validation/platform.go

```diff
@@ -85,11 +85,36 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, fldPath *field
                 if err != nil {
                         ip := net.ParseIP(p.AciNetExt.ClusterSNATDest)
                         if ip == nil {
-                                // error
                                 allErrs = append(allErrs, field.Invalid(fldPath.Child("ClusterSNATDest"), p.AciNetExt.ClusterSNATDest, err.Error()))
                         }
                 }
         }
+
+	machineMask := n.MachineCIDR.Mask
+	if p.AciNetExt.NeutronCIDR.String() != "" {
+                neutronMask := p.AciNetExt.NeutronCIDR.Mask
+                if machineMask.String() != neutronMask.String() {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("NeutronCIDR"),
+					p.AciNetExt.NeutronCIDR.String(), "Machine CIDR and Neutron CIDR have different subnet masks"))
```

Comment:
nit: Here and elsewhere below, let's have a uniform convention when we refer to configuration options that we want to indicate are incorrectly specified. I would suggest that we refer to the exact configuration key, so in the above case the I suggest the message:
"The CIDRs specified in the machineCIDR (%s) and neutronCIDR (%s) configurations in the install-config.yaml have different subnet masks (machineCIDR mask: %s, neutronCIDR mask: %s)"

Reply:
Done